frontend/src/components/core/MainMenu/MainMenu.test.tsx

```diff
@@ -152,4 +171,121 @@ describe("App", () => {
       "About",
     ])
   })
+
+  describe("Onclick deploy button", () => {
+    function testDeployErrorModal(
+      gitInfo: Partial<IGitInfo>,
+      dialogComponent: (module: string) => IDeployErrorDialog
+    ): void {
+      const props = getProps({
+        gitInfo,
+      })
+      const wrapper = mount(<MainMenu {...props} />)
+      const popoverContent = wrapper.find("StatefulPopover").prop("content")
+      // @ts-ignore
+      const menuWrapper = shallow(popoverContent(() => {})).dive()
+
+      const items: any = menuWrapper.prop("items")
+
+      const deployOption = items.find(
+        // @ts-ignore
+        ({ label }) => label === "Deploy this app"
+      )
+
+      deployOption.onClick()
+
+      // @ts-ignore
+      const dialog = dialogComponent(props.gitInfo.module)
+
+      expect(props.showDeployError.mock.calls[0][0]).toStrictEqual(
+        dialog.title
+      )
+      expect(props.showDeployError.mock.calls[0][1]).toStrictEqual(dialog.body)
+    }
+
+    it("no repo or remote", () => {
```

Comment:
Nit: Can we make the "it" messages more descriptive? When you run `yarn test`, it's hard to see what is being tested without looking into the code.